dist/spec/verifiable-payid-protocol-NC.txt

```diff
@@ -0,0 +1,896 @@
+   The "crit" (critical) Header Parameter indicates that extensions to
+   JWS specification are being used that MUST be understood and
+   processed.  This is a required field to be used with "b64" parameter
+   as described in [RFC7797].
+
+3.1.1.6.  jwk
+
+   The "jwk" (JSON Web Key) Header Parameter represents the public key
+   that is used to digitally sign the JOSE header and JWS payload.  This
+   parameter is represented as a JSON Web Key as specified in [RFC7517].
+   In the header above, members of "jwk" represent the properties of the
+   public key, including its value that corresponds to the algorithm
+   "ES256K".
+
+   o  kty: The "kty" (key type) parameter identifies the cryptographic
+      algorithm family used with the key, such as "EC" for Elliptic
+      Curve.
+
+   o  use: The "use" (public key use) parameter identifies the intended
+      use of the public key such as "sig" for signature.
+
+   o  crv : The "crv" (Curve) parameter represents the elliptic curve-
+      type and the hash-type such as "secp256k1" represents curve-type
+      "secp256k1" and the hash-type "SHA-256".
+
+   o  x : Since we assume the "alg" parameter as "ES256K" which is one
+      from the ECDSA family, so "x" parameter represents the
+      X-coordinate of the corresponding public key.
+
+   o  y : Since we assume the "alg" parameter as "ES256K" which is one
+      from the ECDSA family, so "y" parameter represents the
+      Y-coordinate of the corresponding public key.
+
+   Note: "jwk" is one way way of embedding public key in the JOSE
+   header.  For more details on other possible options for "alg" and
+   representing public keys refer to [RFC7515].
+
+3.1.2.  JWS Payload
+
+   The JWS payload is the message that needs to be signed.
+
+
+
+
+
+
+
+
+
+Malhotra & Schwartz     Expires January 31, 2021                [Page 6]
+
+Internet-Draft     Verifiable PayID Protocol Extension         July 2020
+
+
+  {
+          "exp"  : "",
+          "payId": "bob$wallet.com",
+          "payIdAddress": {
+            "expTime": "",
+            "paymentNetwork": "XRPL",
+            "environment": "TESTNET",
+            "addressDetailsType": "CryptoAddressDetails",
+            "addressDetails": {
+              "address": "T7CKYKhRujaxEs9fSxQwJApHsQVPKUgD7EtLWCGTAFBwTha"
+            }
+          }
+  }
+
+3.1.2.1.  exp
+
+   The "exp" field is an optional field as described in [RFC7797].  If
+   used, it SHOULD be set to the expiration time of the cryptographic
+   key used to generate the digital signature.
+
+3.1.2.2.  payId
+
+   The "payId" field is a required field.  The value of "payId" field is
+   the PayID URI in the client request that identifies the payment
+   account information that the JSON object describes.
+
+3.1.2.3.  PayIDAddress
+
+   The "PayIDAddress" is a required field.  The value of "PayIDAddress"
+   field is a JSON object with the following keys:
+
+   o  "expTime": This is an optionl field.  The value of "expTime"
```

Comment:
We should define this more clearly. If it's a String, we should clarify that it's structure (e.g., see ISO-8601). 

However, why not mimic the same structure found in [RFC7519](https://tools.ietf.org/html/rfc7519#section-4.1.4) for the `exp` claim, which is defined to be a `NumericDate`?

Reply:
Agreed, we can use the same structure.